File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    control = {"left": [-1, 0], "right": [1, 0], "up": [0,1], "down": [0,-1]}
    p = [0, 0]
    
    for k in keyinput:
        nx = p[0] + control[k][0]
        ny = p[1] + control[k][1]
        a = board[0]//2
        b = board[1]//2
        if not (-a<=nx<=a and -b<=ny<=b):
            continue
        else:
            p = [nx, ny]
            
    return p

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
class Solution {
     public int[] solution(String[] keyinput, int[] board) {
        int x = 0;
        int y = 0;
        int maxX = (board[0] - 1) / 2;
        int maxY = (board[1] - 1) / 2;

        for (String key : keyinput) {
            switch (key) {
                case "up":
                    if (y + 1 <= maxY) {
                        y++;
                    }
                    break;
                case "down":
                    if (y - 1 >= -maxY) {
                        y--;
                    }
                    break;
                case "left":
                    if (x - 1 >= -maxX) {
                        x--;
                    }
                    break;
                case "right":
                    if (x + 1 <= maxX) {
                        x++;
                    }
                    break;
                default:
                    break;
            }
        }

        return new int[]{x, y};
    }
}